desc/magnetic_fields.py

```diff
@@ -1803,6 +1810,299 @@ def from_surface(
         )
 
 
+class OmnigenousField(Optimizable, IOAble):
+    """Model field with perfect omnigenity.
+
+    Notes
+    -----
+    Doesn't conform to MagneticField API, as it only knows about :math:`|B|` in flux
+    coordinates, not vector B in lab coordinates.
+
+    """
+
+    _io_attrs_ = [
+        "_L_well",
+        "_M_well",
+        "_L_shift",
+        "_M_shift",
+        "_N_shift",
+        "_NFP",
+        "_helicity",
+        "_well_basis",
+        "_shift_basis",
+        "_B_lm",
+        "_x_lmn",
+    ]
+
+    def __init__(
+        self,
+        L_well=0,
+        M_well=2,
+        L_shift=0,
+        M_shift=0,
+        N_shift=0,
+        NFP=1,
+        helicity=(1, 0),
+        B_lm=None,
+        x_lmn=None,
+    ):
+        self._L_well = int(L_well)
+        self._M_well = int(M_well)
+        self._L_shift = int(L_shift)
+        self._M_shift = int(M_shift)
+        self._N_shift = int(N_shift)
+        self._NFP = int(NFP)
+        self.helicity = helicity
+        self._well_basis = ChebyshevPolynomial(L=self.L_well)
```

Comment:
this is only the radial basis right? the `basis` in the eta direction is the splines? maybe make this `_well_radial_basis` to be clearer

Reply:
I like to think of it as the full basis for the `B_lm` parameters, it's just that the spline part doesn't have any modes to keep track of. The users shouldn't be interacting with the basis directly anyways.